controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -479,6 +457,23 @@ func (r *StorageConsumerReconciler) reconcileConsumerConfigMap(
 			); err != nil {
 				return err
 			}
+
+			// update desired state of configmap
+			if consumerConfigMap.Data == nil {
+				consumerConfigMap.Data = map[string]string{}
+			}
+
+			defaultConsumerResourceNames := util.GetStorageConsumerDefaultResourceNames(
+				r.storageConsumer.Name,
+				string(r.storageConsumer.UID),
+				availableServices,
+			)
+			for key := range defaultConsumerResourceNames {
+				consumerConfigMap.Data[key] = cmp.Or(
+					strings.Trim(consumerConfigMap.Data[key], " "),
+					defaultConsumerResourceNames[key],
+				)
+			}
```

Comment:
This works for the first reconciliation if there is no controllerRef. What if, as a day-2 operation, I enable the NFS service, in this case the NFS keys/values won't exist in the configmap as the controllerRef is not nil. 

We need to move this below the OwnerRef/ControllerRef and check if the current consumer if the controllerRef we update the values